### DISCUSSION TITLE
How to renew certificate?

### DISCUSSION THREAD

**Discussion by mariusa** [piku/piku#310] [Q&A] [answered]
Hi,

I've discovered via Let's Encrypt Expiry Bot that an app certificate will expire in 14 days. I've searched this repo for renewal but couldn't find how to renew it.

With the standard piku setup, including acme.sh as instructed, how is the certificate renewal handled?
Does the app need to be manually restarted for some checks & renewal to kick in?

Thanks

**Comment by chr15m** [comment, marked as answer]
acme.sh installs a cron job which should auto renew the certificates when they are due. If you `su piku` and then `crontab -l` you can see the crontab. `tail .acme.sh/acme.sh.log -n 100` will show you the last 100 lines of the acme log. You can use this to see if there are any errors renewing.

**Reply by mariusa** [reply]
Thanks, it helped!

Log shown `Skip invalid cert for: host.com`

```
acme.sh --list # one of the certs had KeyLength empty
acme.sh --renew -d host.com --force --ecc # solved
```
